File: day6.rb
def generation(shoal)
  x = shoal.shift()
  shoal.push(0)
  shoal[6] += x
  shoal[8] += x
end

shoal = [0] * 10
total = 0

File.open(ARGV[0]).read.strip.split(',').each do |fish|
  shoal[fish.to_i] += 1
  total += 1
end
for gen in 1..80
  total += generation(shoal)
end
puts "Part 1: #{total}"
for gen in 81..256
  total += generation(shoal)
end
puts "Part 2: #{total}"

File: day6.py
#!/usr/bin/env python3

import sys


def generation(shoal):
    spawn = shoal.pop(0)
    shoal.append(0)
    shoal[6] += spawn
    shoal[8] += spawn


def main():
    with open(sys.argv[1], 'r') as fhan:
        line = fhan.readline().strip().split(',')
    start = [int(x) for x in line]
    shoal = [0] * 9
    for fish in start:
        shoal[fish] += 1

    for i in range(80):
        generation(shoal)
    print("Part 1", sum(shoal))
    for _ in range(256 - 80):
        generation(shoal)
    print("Part 2", sum(shoal))


if __name__ == '__main__':
    main()
